string[] vec1
string[] vec2